src/frontends/lean/parser.cpp

```diff
@@ -533,6 +533,10 @@ void parser::add_local_level(name const & n, level const & l, bool is_variable)
 }
 
 void parser::add_local_expr(name const & n, expr const & p, bool is_variable) {
+    if (!m_in_quote) {
+        // certainly not in a pattern
+        m_in_pattern = false;
```

Comment:
It looks fine to me. Could you please add a comment explaining the motivation for the hack?
I regularly forget the motivations for the hacks I have added. 
So, I'm trying to comment them before I forget them :)